root/edit/details/EditPlace.js

```diff
@@ -0,0 +1,131 @@
+/*
+ * @flow
+ * Copyright (C) 2019 Anirudh Jain
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import DescriptiveLink from
+  '../../static/scripts/common/components/DescriptiveLink';
+import Diff from '../../static/scripts/edit/components/edit/Diff';
+import FullChangeDiff from
+  '../../static/scripts/edit/components/edit/FullChangeDiff';
+import FullChangeDiffContent from
+  '../../static/scripts/edit/components/edit/FullChangeDiffContent';
+import {formatCoordinates} from '../../utility/coordinates';
+import formatDate from '../../static/scripts/common/utility/formatDate';
+import yesNo from '../../static/scripts/common/utility/yesNo';
+
+type PlaceEditT = {
+  ...EditT,
+  display_data: {
+    address: CompT<string>,
+    area: CompT<AreaT>,
+    begin_date: CompT<PartialDateT>,
+    comment: CompT<string>,
+    coordinates: CompT<CoordinatesT>,
+    end_date: CompT<PartialDateT>,
+    ended: CompT<boolean>,
+    name: CompT<string>,
+    place: PlaceT,
+    type: CompT<PlaceTypeT>,
+  },
+};
+
+const EditPlace = ({edit}: {edit: PlaceEditT}) => {
+  const display = edit.display_data;
+  const place = display.place;
+  const oldType = display.type ? display.type.old : {name: ''};
+  const newType = display.type ? display.type.new : {name: ''};
+  const notDiffContent = display.area
+    ? display.area.new.gid === display.area.old.gid : false;
+  return (
+    <table className="details edit-place">
+      <tbody>
+        <tr>
+          <th>{addColonText(l('Place'))}</th>
+          <td><DescriptiveLink entity={place} /></td>
+        </tr>
+        {display.name ? (
+          <Diff
+            label={addColonText(l('Name'))}
+            newText={display.name.new}
+            oldText={display.name.old}
+            split="\s+"
+          />
+        ) : null}
+        {display.comment ? (
+          <Diff
+            label={addColonText(l('Disambiguation'))}
+            newText={display.comment.new}
+            oldText={display.comment.old}
+            split="\s+"
+          />
+        ) : null}
+        {display.type ? (
+          <FullChangeDiff
+            label={addColonText(l('Type'))}
+            newText={newType.name}
+            oldText={oldType.name}
+          />
+        ) : null}
+        {display.address ? (
+          <Diff
+            label={addColonText(l('Address'))}
+            newText={display.address.new}
+            oldText={display.address.old}
+            split="\s+"
+          />
+        ) : null}
+        {display.area ? (
+          <FullChangeDiffContent
+            label={addColonText(l('Area'))}
+            newContent={display.area.new
+              ? <DescriptiveLink entity={display.area.new} />
+              : ''}
+            notDiffContent={notDiffContent}
```

Comment:
We currently don't need a prop like this to control the visibility, since `display.area` won't exist unless the area actually changed (so this would never be false).

I'm guessing you might need this sort of behavior for your in-page edit previews, though? (Assuming it passes in old/new data without checking them for equality.) Even in that case, we still don't need this prop, because we can just perform the check above where `display.area ?` is, with something like `(display.area && display.area.new?.gid === display.area.old?.gid) ?`.

Reply:
Don't know if this is still necessary now 🤔 removed the component altogether